### ISSUE TITLE
Using field_transformer option in CTGAN model gives better results than explicitly transforming data outside CTGAN model

### ISSUE THREAD

**Issue by PJPRoche** [sdv-dev/SDV#1172]
### Environment Details

* SDV version: 0.17.0
* Python version: 3.8
* Operating System: Windows / conda env

### Error Description

Specifying the field_transformers and passing into the CTGAN model gives quite different results to using RDT transformers and passing in the transformed data to the CTGAN model. 

### Steps to reproduce

```
import pandas as pd

from sdmetrics.reports.single_table import QualityReport
from sdv.tabular import CTGAN

from rdt.transformers.categorical import LabelEncoder
from rdt.transformers.numerical import FloatFormatter
from rdt import HyperTransformer


NUM_ROWS = 100000
NUM_EPOCHS = 100

real_data = pd.read_csv(<<<PATH_TO_CSV_FILE>>>\real_data.csv")

field_types = {
    "COL_A": {"subtype": "float", "type": "numerical"},
    "COL_B": {"subtype": "float", "type": "numerical"},
    "COL_C": {"type": "categorical"},
}

field_transformers = {
    "COL_A": "FloatFormatter",
    "COL_B": "FloatFormatter",
    "COL_C": "LabelEncoder_noised",
}

METADATA = {
    "fields": field_types
}


# ==============================================================
# Run model using auto-transformer
model = CTGAN(
    epochs=NUM_EPOCHS,
    verbose=True,
    field_types=field_types,
    field_transformers=field_transformers,
)
model.fit(real_data)

# Get Synthetic data
synthetic_data = model.sample(NUM_ROWS)

report = QualityReport()
report.generate(real_data, synthetic_data, METADATA)

ds_report = report.get_details(property_name="Column Shapes")
print(ds_report)

# ensure model is deleted
del model
del report

# ==============================================================
# Run model but using explicit transformer

ht = HyperTransformer()

config = {
    "sdtypes": {
        "COL_A": "numerical",
        "COL_B": "numerical",
        "COL_C": "categorical",
    },
    "transformers": {
        "COL_A": FloatFormatter(missing_value_replacement='mean'),
        "COL_B": FloatFormatter(missing_value_replacement='mean'),
        "COL_C": LabelEncoder(add_noise=True),
    }
}

ht.set_config(config=config)

# transform real data
transformed_real_data = ht.fit_transform(data=real_data)

# Run model using explicit transformer
model = CTGAN(
    epochs=NUM_EPOCHS,
    verbose=True,
)
model.fit(transformed_real_data)

# Get Synthetic data
synthetic_data = model.sample(NUM_ROWS)
# reverse transform the data
synthetic_data = ht.reverse_transform(synthetic_data)

report = QualityReport()
report.generate(real_data, synthetic_data, METADATA)

ds_report = report.get_details(property_name="Column Shapes")
print(ds_report)
```

In the first case, I see:

```
  Column        Metric  Quality Score
0  COL_A  KSComplement        0.89063
1  COL_B  KSComplement        0.90466
2  COL_C  TVComplement        0.95165
```

and in the second I see:

```
   Column        Metric  Quality Score
0  COL_A  KSComplement        0.95602
1  COL_B  KSComplement        0.90111
2  COL_C  TVComplement        0.79928
```

I would have expected to see more consistent results between the two approaches. I have seen the same on other data sets and it always seems to the COL_C (categorical) variable is "worse" when explicitly using the RDT transformer.

I have not had the time yet to look into this more and compare data at the point is is passed to the final CTGAN model and training starts. Unless there is some seeding and difference how the Label encoder chooses values (I have not looked into this yet either), I would have expected the two approaches to yield more similar results

[real_data.csv](https://github.com/sdv-dev/SDV/files/10393690/real_data.csv)

**Comment by npatki** [comment]
Hi @PJPRoche thanks the detailed code and for attaching the data! We'll try to replicate and investigate what's happening.

In the meantime, could you double check the versions of both the SDV and SDMetrics libraries that you have in your environment? I see above you've listed the SDV version as `0.17.0`. However, this earlier version of SDV is incompatible with SDMetrics versions `0.7.0+`, which is needed to generate the report.

**Comment by npatki** [comment]
Hi @PJPRoche -- A quick update from our investigations. It appears that this problem is due to a rounding bug, filed in [RDT Issue 556](https://github.com/sdv-dev/RDT/issues/556) that should be resolved in the next RDT release (`1.3.0`).

### Workaround

Until the update is out, I was able to fix the issue if I rounded the transformed data to 14 digits. 

```python
# transform real data
transformed_real_data = ht.fit_transform(data=real_data)

# Run model using explicit transformer
model = CTGAN(
    epochs=NUM_EPOCHS,
    verbose=True,
)

# FIX: Fit the rounded data
model.fit(transformed_real_data.round(14))
```

I verified that with this fix, the `TVComplement` score was ultimately `0.95423`

### Technical Details
1. When using an explicit transformer, the RDT's `LabelEncoder` creates a noised column with over 14 decimal digits
2. You are feeding the column with >14 digits into a CTGAN model
3. The CTGAN is creating its own RDT HyperTransformer and feeding it the fully-numerical data
4. RDT encounters a bug when a numerical column has over 14 decimal digits

You can see this happening live. You'll see that the CTGAN model is generating whole numbers, not decimals, for `COL_C`. This ultimately affects the quality.

**Comment by npatki** [comment]
Hi @PJPRoche, I'm closing this issue off since we found the root cause of the error -- and will defer to #1039 for the final fix.

If you are still observing any differences, feel free to reply and I can reopen the issue for investigation.
